extensions/2.0/Khronos/KHR_interactivity/Specification.adoc

```diff
@@ -0,0 +1,2914 @@
+
+Behavior graphs and trigger-action lists share common features, such as being safe and sandboxed, offering limited execution models controlled by the viewer, and both supporting the “trigger” and “action” node categories. However, there are also significant differences between the two models. Trigger-action lists lack “Queries”, “Logic”, and “Control Flow” nodes, meaning that sophisticated behavior based on queries, logic, or control flow branches is not possible. This lack of functionality greatly affects the ability to create complex behavior and control structures and rules out the implementation of advanced control flow structures in the future.
+
+On the other hand, behavior graphs are a superset of trigger-action lists, meaning that the former can support everything that trigger-action lists can, and more. Behavior graphs support “Queries”, “Logic” and “Control Flow” nodes, making them more expressive and capable of creating more sophisticated behaviors. This makes behavior graphs the preferred method of choice for high-end game engines, as it offers an identical safety model as trigger-action lists while being more expressive.
+
+=== Turing Completeness
+The execution model and node choices for this extension mean that it is Turing-complete. This means that an implementation of this can execute any computation and it is also hard to predict if it will run forever (e.g. halt or not.)
+
+While this may present security implications, it is not a major hindrance and can be safely mitigated so that any implementation does not become susceptible to denial of services by badly behaving behavior graphs, whether intention or not.
+
+The main way to mitigate the risk of non-halting behavior graphs is to limit the amount of time given to them for execution, both in terms of individual time slice as well as overall execution time.
+
+=== Implementation Limitations
+There will be limitations in engines and devices to such as:
+
+* Number of nodes in the graph
+* Number of variables
+* Number of custom events
+* Number of concurrent pending events/async nodes
+* Number of nodes executed per time slice
+* Speed of graph execution
+
+These limitations are not defined in this specification.
+
+
+[[concepts]]
+= Concepts
+
+[[concepts-general]]
+== Graphs
+
+A behavior graph is a JSON object containing _nodes_. It **MAY** also contain custom variables and custom events.
+
+Behavior graphs are directed graphs with no directed cycles.
+
+[[nodes]]
+== Nodes and Sockets
+
+A _node_ is a JSON object, which represents an executable item. Each node is defined by its _type_ and a set of _sockets_. There are four kinds of sockets.
```

Comment:
This is kind of a nitpick, and I've been debating whether or not to bring this up, but I figure I should.

I don't like that elements of the KHR_interactivity behavior graph are called "nodes". We already use the word "node" for something *very specific* in glTF [here](https://github.com/KhronosGroup/glTF/blob/main/specification/2.0/schema/node.schema.json). It would be more confusing than necessary to overload this word with another meaning.

Instead, I recommend changing the name to "block", short for "script block". It is common in visual scripting systems to call their elements "visual script blocks". The term "block" is common, descriptive, and most importantly, is not already used for something else in glTF. It would be a clearer term to use.